From 5b5b5ef3d4eb807ed76bfdc92106ff0149785e1d Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Wed, 16 Nov 2011 10:13:24 +0800
Subject: [PATCH 211/502] ENGR00162320 the keyguard is not shown when
 suspend/resume 10%

keyguard is drawn when early suspend,
when resume,the surfaceflinger will draw the screen, but blocked
by the hwc.

Signed-off-by: b02247 <b02247@freescale.com>
---
 mx5x/hwcomposer/hwcomposer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mx5x/hwcomposer/hwcomposer.cpp b/mx5x/hwcomposer/hwcomposer.cpp
index de2e673..d82802c 100755
--- a/mx5x/hwcomposer/hwcomposer.cpp
+++ b/mx5x/hwcomposer/hwcomposer.cpp
@@ -503,7 +503,7 @@ static int hwc_set(hwc_composer_device_t *dev,
 #endif
 		//HWCOMPOSER_LOG_RUNTIME("==============hwc_set=2==============\n");
 #if 1 
-    if((ctx == NULL) || (ctx && ctx->ui_refresh)) {
+    if((ctx == NULL) || (ctx && (ctx->ui_refresh || (!ctx->ui_refresh && !ctx->vd_refresh )))) {
         EGLBoolean sucess = eglSwapBuffers((EGLDisplay)dpy, (EGLSurface)sur);
         if (!sucess) {
             return HWC_EGL_ERROR;
-- 
1.8.0

